rollbar-reactive-streams-reactor/src/test/java/com/rollbar/reactivestreams/notifier/sender/http/AsyncHttpClientFactoryTest.java

```diff
@@ -0,0 +1,13 @@
+package com.rollbar.reactivestreams.notifier.sender.http;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+public class AsyncHttpClientFactoryTest {
+  @Test
+  public void ifReactorIsInClasspathItShouldReturnReactorClient() {
+    assertThat(AsyncHttpClientFactory.defaultClient(), Matchers.instanceOf(ReactorAsyncHttpClient.class));
+  }
```

Comment:
The negative test is not present for very annoying logistical reasons, setting a classpath and forcing a specific test to reliably run with that classpath is quite a hassle. 

The only negative case we have is in  the `rollbar-java-reactivestreams` project, where the tests run in a classpath that doesn't include the reactor http dependency, and the resulting factory is different. 

This is "the java way", a questionable practice of letting the presence of a class in the runtime classpath modify the behaviour of the application, without the user writing any code, but so much convenience has been built around this practice that it's hard to go against it.

Reply:
Fair enough!

Anyway, why don't you add a test for this:

> The only negative case we have is in the `rollbar-java-reactivestreams` project, where the tests run in a classpath that doesn't include the reactor http dependency, and the resulting factory is different.

I would expect to test the other results from the factory.

Reply:
That test exists, but is in a different project, since its classpath is set up in such a way that it produces what we would consider the negative result for _this_ test. 

As mentioned in another comment, there is no convenient way to test classpath related logic, despite being commonly used in Java. But if this represents a big problem we can do the work, though I think compromises in test coverage are necessary to avoid overspending effort on tests.

Reply:
Not a big problem. Fair enough to skip some edge tests in this case.